Makefile

```diff
@@ -125,6 +133,15 @@ toc-verify:
 vet: ## Run go vet against code.
 	$(GO_CMD) vet ./...
 
+## Create Kind Cluster for E2E
+.PHONY: kind-create
+kind-create:
+	kind create cluster --image=$(E2E_KIND_VERSION)
```

Comment:
I have no idea whether the test environment is isolated. If not, the default name of kind might be conflict, cc @BenTheElder who may give us the answer :)

Reply:
I expected things won't work exactly right once this PR gets merged.  I wanted to get something merged in this repo and then test the test-infra PR to make sure everything looks good.  If there is an issue, I imagine we will find it out pretty quick.  But I do appreciate @BenTheElder taking a look if possible.

Reply:
Uh "isolated" is a strong word when you're sharing a host kernel with what is already a kubernetes node. But in Prow if you're running kind the host nodes run containerd and we docker in containerd and then the kind nodes run containerd in docker. It's a leaky abstraction but the cluster name is not going to conflict with another docker container because there's a dockerd for each job.

On GitHub actions iirc the isolation is a bit stronger and similarly there's a docker instance for each workflow

Reply:
You may still want to consider a unique name in CI scripts for local runs.

Reply:
Your makefile could export KIND_CLUSTER_NAME to avoid having to plumb it everywhere.

Reply:
I'll add a KIND_CLUSTER_NAME as not deleting our dev default kind environment is probably a good move.